docs/using/polar-syntax.rst

```diff
@@ -285,25 +285,35 @@ will compare the value of the variable age with 10 and unify if it's less than 1
 Cut
 ^^^
 
-The ``cut`` operator the query engine to the enclosing rule definition,
-and refuses to consider any others. Rules that have already run are not
-"un-run", though, or avoided by using cut; it just ensures that no
-*others* will run. Such "other" rule definitions are often less specific
-rules, and `cut` can be used, e.g., to override an ``allow`` rule for a
-less specific class.
-
-``cut`` can appear anywhere in a rule body, but terms that
-proceed it must succeed in order for it to be reached, so it
-frequently appears at the end of the body: **if** so-and-so is true,
-then **cut** out all other alternatives.  ``cut`` should be
-used sparingly.
+By default, Polar runs all of the definitions for a given rule that are
+applicable to the given set of arguments (i.e., whose specializers are
+matched). The ``cut`` operator overrides this behavior by *committing* to
+the enclosing rule definition: the query engine will not run any others.
+Rule definitions that have already run are not "un-run", though, or avoided
+by using cut; it just ensures that no *others* will run.
+
+Because Polar runs rules in most-to-least-specific order, these "other"
+rule definitions are always *less specific* than the current one; i.e.,
+they may have specializers that are superclasses (and therefore less specific)
+of those in the current rule. This allows ``cut`` to override a rule that
+is specialized on a less specific class. You can think of ``cut`` as a sort
+of dual to ``super()`` in other object-oriented languages (e.g., Python):
```

Comment:
```suggestion
of dual to ``super()`` in object-oriented languages (e.g., Python):
```

Reply:
No, Polar is an object-oriented language.

Reply:
you're an object-oriented language